src/System.Globalization.Calendars/tests/CalendarHelpers.cs

```diff
@@ -95,84 +216,105 @@ public static void GetWeekOfYear_Invalid(Calendar calendar)
+
+            Assert.Throws<ArgumentOutOfRangeException>(() => calendar.ToDateTime(minYear, minDateTime.Month, minDateTime.Day - 1, minDateTime.Hour, minDateTime.Minute, minDateTime.Second, minDateTime.Millisecond, minEra));
+            Assert.Throws<ArgumentOutOfRangeException>(() => calendar.ToDateTime(maxYear, maxDateTime.Month, maxDateTime.Day + 1, maxDateTime.Hour, maxDateTime.Minute, maxDateTime.Second, maxDateTime.Millisecond, maxEra));
+
+            Assert.Throws<ArgumentOutOfRangeException>(() => calendar.ToDateTime(minYear, minDateTime.Month, minDateTime.Day, minDateTime.Hour - 1, minDateTime.Minute, minDateTime.Second, minDateTime.Millisecond, minEra));
+            Assert.Throws<ArgumentOutOfRangeException>(() => calendar.ToDateTime(maxYear, maxDateTime.Month, maxDateTime.Day, maxDateTime.Hour + 1, maxDateTime.Minute, maxDateTime.Second, maxDateTime.Millisecond, maxEra));
+
+            Assert.Throws<ArgumentOutOfRangeException>(() => calendar.ToDateTime(minYear, minDateTime.Month, minDateTime.Day, minDateTime.Hour, minDateTime.Minute - 1, minDateTime.Second, minDateTime.Millisecond, minEra));
+            Assert.Throws<ArgumentOutOfRangeException>(() => calendar.ToDateTime(maxYear, maxDateTime.Month, maxDateTime.Day, maxDateTime.Hour, maxDateTime.Minute + 1, maxDateTime.Second, maxDateTime.Millisecond, maxEra));
+
+            Assert.Throws<ArgumentOutOfRangeException>(() => calendar.ToDateTime(minYear, minDateTime.Month, minDateTime.Day, minDateTime.Hour, minDateTime.Minute, minDateTime.Second - 1, minDateTime.Millisecond, minEra));
+            Assert.Throws<ArgumentOutOfRangeException>(() => calendar.ToDateTime(maxYear, maxDateTime.Month, maxDateTime.Day, maxDateTime.Hour, maxDateTime.Minute, maxDateTime.Second + 1, maxDateTime.Millisecond, maxEra));
+
+            Assert.Throws<ArgumentOutOfRangeException>(() => calendar.ToDateTime(minYear, minDateTime.Month, minDateTime.Day, minDateTime.Hour, minDateTime.Minute, minDateTime.Second, minDateTime.Millisecond - 1, minEra));
+            Assert.Throws<ArgumentOutOfRangeException>(() => calendar.ToDateTime(maxYear, maxDateTime.Month, maxDateTime.Day, maxDateTime.Hour, maxDateTime.Minute, maxDateTime.Second, maxDateTime.Millisecond + 1, maxEra));
         }
 
+        [Theory]
+        [MemberData(nameof(Calendars_TestData))]
         public static void ToFourDigitYear_Invalid(Calendar calendar)
         {
             Assert.Throws<ArgumentOutOfRangeException>("year", () => calendar.ToFourDigitYear(-1));
-            Assert.Throws<ArgumentOutOfRangeException>("year", () => calendar.ToFourDigitYear(calendar.MinSupportedDateTime.Year + YearOffset(calendar) - 1));
-            Assert.Throws<ArgumentOutOfRangeException>("year", () => calendar.ToFourDigitYear(calendar.MaxSupportedDateTime.Year + YearOffset(calendar) + 1));
+            // JapaneseCalandar allows any inputs below the max year
+            if (!(calendar is JapaneseCalendar))
```

Comment:
Okay folks: this is implementational, not to do with my test code.
If you take a look at the JapaneseCalendar source code: https://github.com/dotnet/coreclr/blob/master/src/mscorlib/corefx/System/Globalization/JapaneseCalendar.cs#L302

The implementation for ToFourDigitYear only checks if year <= 0 or year >= max not anything else.